library/run_cmds.py

```diff
@@ -102,13 +101,14 @@ def __init__(self, module):
         if isinstance(self.fact_names, list):
             for f_name in self.fact_names:
                 f_name_list = f_name.strip().split('_', 1)
-                command = DEFAULT_CMD_DICT[f_name_list[0]]
-                fact = f_name_list[1]
-                if command in DEFAULT_CMDS:
-                    if command in self.facts_requested.keys():
-                        self.facts_requested[command].append(fact)
-                    else:
-                        self.facts_requested[command] = [fact]
+                if f_name_list[0] in DEFAULT_CMD_DICT:
+                    command = DEFAULT_CMD_DICT[f_name_list[0]]
+                    fact = f_name_list[1]
+                    if command in DEFAULT_CMDS:
```

Comment:
Also proper fact validation will be handled with this issue: https://github.com/quipucords/rho/issues/55

And should be pretty easy with `utilities.DEFAULT_FACTS_TUPLE`.